### PR TITLE
Spring security

### PR DESCRIPTION
**Pull request by harluss**
Spring Security email and password based authentication with JWT (backend):
- Added Spring Security configuration
- Added Swagger authentication configuration
- Added password encryption
- Wired up getUser by id to AWS Service
- Added access and refresh tokens logic (AuthTokenUtility and AuthFilter) and endpoints (AuthController)
- Expanded global error handling
- Updated UserService tests
- Added IT tests for AuthController**
- Updated Readme
- Updated docker-compose
- Added dummy data seeding script

** Tests check existing auth endpoints only atm, since all routes are opened for now. More security tests will be added, once React app is expanded to handle user login flow and tokens logic and certain routes are locked behind authentication.

NOTE:
This is basic email and password authentication flow, doesn't handle any roles or authorities as we don't have those in the app yet, but some of the logic to back it up is there and can be expanded.
Some form of white or blacklisting refresh token will be added in the future as well.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -49,13 +49,36 @@
...
+      <groupId>io.jsonwebtoken</groupId>
+      <artifactId>jjwt</artifactId>
+      <version>0.9.1</version>
+    </dependency>
+
+    <!-- jaxb-api -->
+    <dependency>
+      <groupId>javax.xml.bind</groupId>
+      <artifactId>jaxb-api</artifactId>
+      <version>2.3.1</version>
```
Comment (rjclancy) [inline pom.xml:68]: move versions to properties

File: src/main/java/com/cor/devsquare/config/SecurityConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+  @Override
+  protected void configure(AuthenticationManagerBuilder auth) throws Exception {
+    auth.userDetailsService(userService)
+        .passwordEncoder(passwordEncoder());
+  }
+
+  @Override
+  @Bean
+  public AuthenticationManager authenticationManagerBean() throws Exception {
```
Comment (rjclancy) [inline src/main/java/com/cor/devsquare/config/SecurityConfiguration.java:71]: What was the reason you need to override AuthenticationManager?

**Review by rjclancy — APPROVED** [review]

File: src/main/java/com/cor/devsquare/config/SecurityConfiguration.java (hunk 2 above)
Comment (harluss) [reply src/main/java/com/cor/devsquare/config/SecurityConfiguration.java:71]: Something to do with changes in never Spring Boot versions if I remember correctly. This is needed in Spring Boot >2.SOMETHING

File: pom.xml (hunk 1 above)
Comment (harluss) [reply pom.xml]: done

**Pull request merged**